src/telemetry.ts

```diff
@@ -29,23 +30,29 @@ export class Telemetry {
     }
   }
 
-  async initialize() {
-    try {
-      this.api = await vscode.commands.executeCommand(
-        "ruby-lsp.getPrivateTelemetryApi"
-      );
-    } catch (_error) {
-      // Do nothing if no telemetry api is available
+  async sendEvent(event: TelemetryEvent) {
```

Comment:
This approach makes sense to me overall, but I have a question - if we're moving the call to `initialize` inside the `sendEvent` method, why do we need to check for the telemetry API five times? Shouldn't all the APIs be loaded at this point if they exist? (Or maybe I'm misunderstanding the order of operations.)

Reply:
My idea was to not keep trying to execute the command forever, if someone is using the Ruby LSP without a private metrics plugin attached.

We know that the command will be loaded within 5 tries, so if we can't find the command after that, it's probably because the user doesn't have a metrics plugin - and not because it hasn't been loaded yet.

Reply:
Can't we check in `vscode.extensions.all` if the extension is there instead and keep trying if we found it? Or maybe this list would not be up-to-date if the extension is not loaded?

Reply:
That would remove the ability of other developers/companies adding their own private metrics plugin, as we wouldn't know the extension name or identifier.

Reply:
With the current approach, isn't there a change we're going to reach the limit before the other extension loads? What if I have 5 tabs opened already? Wouldn't I risk sending 5 `didOpen` events before the other extension loads?

Reply:
> Maybe we should just try to execute the command every time?

Would that be a noticeable overhead? Do we have a way to measure it?

Reply:
I don't think so. That's what Sorbet does and it doesn't seem to have a big impact. I'll switch the implementation.